tsl/src/continuous_aggs/drop.c

```diff
@@ -15,9 +15,38 @@
 
 #include "create.h"
 
+/* drop chunks from the materialization hypertable that fall within the time
+ * range.
+ */
+static void
+cagg_drop_mat_chunks(Oid mattable_relid, Datum older_than_datum, Datum newer_than_datum,
```

Comment:
I wonder why we don't simply call `ts_chunk_do_drop_chunks` here. Right now that doesn't support calls on continous agg mat tables, but that should actually also be fixed. And making this call that function would unify all the checks.

Reply:
The table_name parameter is optional for drop_chunks. The variant which can be called without a table_name can lead to inadvertent errors if we allow dropping materialization tables.
`SELECT drop_chunks(schema_name => 
    table_name => ,
    newer_than => -20,
    verbose => true);
`
with this variant everything works fine when a)  the user calls it explicitly on a materialization table OR b) calls it explicitly on a raw hypertable (cascade_to_materialization parameter decides what to do with the materialization hypertable).

Now consider this variant,
`
SELECT drop_chunks(
    newer_than => -20,
    verbose => true);
`
Here a user can *inadvertently drop materialization tables* . cascade_to_materialization parameter has no effect here i(if we allow drop chunks directly on the mat hypertable) and so the user is not forced to make a choice. Easy to go wrong with this call.
 
I think this variant of drop_chunks should actually be removed. (can this be done for 1.7 ?) The table_name parameter should be mandatory. This would be a cleaner interface and also maps to our drop_chunks policy.
We should enable drop_chunks on materialization hypertable only using the modified interface (where table_name is mandatory) to avoid unexpected issues with dropping chunks from materialized hypertables.